### ISSUE TITLE
doc loading and generation crashing my OS

### ISSUE THREAD

**Issue by a3nima** [h2oai/h2ogpt#490]
I think it crashes my graphics driver and then OS reboots..

win10, nvidia 3090, driver 531.79, newest h2ogpt build (git pull), installed accordingly to readme file. run in document q/a mode, started without errors.

my start parameters:

> python generate.py --base_model=h2oai/h2ogpt-gm-oasst1-en-2048-falcon-7b-v3 --langchain_mode=MyData --score_model=None --load_8bit=True

start log after crash:

```
Already up to date.
Using Model h2oai/h2ogpt-gm-oasst1-en-2048-falcon-7b-v3
Prep: persist_directory=db_dir_UserData does not exist, regenerating
Chose UserData but user_path is empty/None
Removing old MyData: /tmp\db_dir_MyData_67dde955-7a7f-4d3d-a775-117fbcaaee82
Starting get_model: h2oai/h2ogpt-gm-oasst1-en-2048-falcon-7b-v3
device_map: {'': 0}
bin e:\TEXT-AI\envs\h2ogpt\lib\site-packages\bitsandbytes\libbitsandbytes_cuda118.dll
Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████| 2/2 [00:08<00:00,  4.33s/it]
Model {'base_model': 'h2oai/h2ogpt-gm-oasst1-en-2048-falcon-7b-v3', 'tokenizer_base_model': '', 'lora_weights': '', 'inference_server': '', 'prompt_type': 'prompt_answer', 'prompt_dict': {'promptA': '', 'promptB': '', 'PreInstruct': '<|prompt|>', 'PreInput': None, 'PreResponse': '<|answer|>', 'terminate_response': ['<|prompt|>', '<|answer|>', '<|endoftext|>'], 'chat_sep': '<|endoftext|>', 'chat_turn_sep': '<|endoftext|>', 'humanstr': '<|prompt|>', 'botstr': '<|answer|>', 'generates_leading_space': False}}
```

bsod files and codes:
```
dxgkrnl.sys	dxgkrnl.sys+2665be	fffff806`74b00000	fffff806`74eaa000	0x003aa000	0x0f2ddf59							
ntoskrnl.exe	ntoskrnl.exe+d25440	fffff806`65600000	fffff806`66646000	0x01046000	0xa62a1a64	05.05.2058 00:32:04						
nvlddmkm.sys	nvlddmkm.sys+1073ac0	fffff806`82420000	fffff806`85cb8000	0x03898000	0x644818e5	25.04.2023 20:16:05						
```

if it does not bsod, it crashes with CUDA errors like this:
```
File "e:\TEXT-AI\envs\h2ogpt\lib\site-packages\InstructorEmbedding\instructor.py", line 278, in forward
assert torch.sum(attention_mask[local_idx]).item() >= context_masks[local_idx].item(),
RuntimeError: CUDA error: unspecified launch failure
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with TORCH_USE_CUDA_DSA to enable device-side assertions.
```

**Comment by pseudotensor** [comment]
FYI my windows system is windows 10 pro with 3090Ti and the pythia models were no trouble.  I can try falcon.

**Comment by pseudotensor** [comment]
It might be running out of GPU memory?  Maybe try changing to the other embedding if the instructor embedding is crashing?

**Comment by pseudotensor** [comment]
I went through entire windows documentation and updated it, and confirmed GPU working etc. Please try updated docs: https://github.com/h2oai/h2ogpt/blob/main/docs/README_WINDOWS.md

**Comment by a3nima** [comment]
> It might be running out of GPU memory? Maybe try changing to the other embedding if the instructor embedding is crashing?

out of memory not, except the virtual memory of the system is not big enough, VRAM it was only using 10-12GB when crashed  (i got 24).

**Comment by a3nima** [comment]
I followed the newly updated install page for windows, but still get the error when trying to add a pdf:
To create a public link, set `share=True` in `launch()`.
100%|████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  5.10it/s]
0it [00:00, ?it/s]
load INSTRUCTOR_Transformer
max_seq_length  512
Traceback (most recent call last):
  File "E:\TEXT-AI\h2ogpt\src\gradio_runner.py", line 2691, in update_user_db
    return _update_user_db(file, db1s=db1s, chunk=chunk, chunk_size=chunk_size,
  File "E:\TEXT-AI\h2ogpt\src\gradio_runner.py", line 2867, in _update_user_db
    db = get_db(sources, use_openai_embedding=use_openai_embedding,
  File "E:\TEXT-AI\h2ogpt\src\gpt_langchain.py", line 113, in get_db
    db = Chroma.from_documents(documents=sources,
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\langchain\vectorstores\chroma.py", line 564, in from_documents
    return cls.from_texts(
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\langchain\vectorstores\chroma.py", line 528, in from_texts
    chroma_collection.add_texts(texts=texts, metadatas=metadatas, ids=ids)
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\langchain\vectorstores\chroma.py", line 166, in add_texts
    embeddings = self._embedding_function.embed_documents(list(texts))
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\langchain\embeddings\huggingface.py", line 158, in embed_documents
    embeddings = self.client.encode(instruction_pairs, **self.encode_kwargs)
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\InstructorEmbedding\instructor.py", line 539, in encode
    out_features = self.forward(features)
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\torch\nn\modules\container.py", line 217, in forward
    input = module(input)
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "e:\TEXT-AI\venv\h2ogpt\lib\site-packages\InstructorEmbedding\instructor.py", line 278, in forward
    assert torch.sum(attention_mask[local_idx]).item() >= context_masks[local_idx].item(),\
RuntimeError: CUDA error: an illegal memory access was encountered
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

**Comment by pseudotensor** [comment]
`CUDA error: an illegal memory access was encountered` is a pretty bad error.  Either OOM on GPU or host or some other setup issue.  It's a very generic error as well, not related to h2oGPT specifically.

it's similar to what you see here "RuntimeError: CUDA error: unspecified launch failure" https://github.com/h2oai/h2ogpt/issues/462#issuecomment-1637138424

Also this "RuntimeError: CUDA error: invalid program counter"

All some common issue that's not directly related to h2oGPT.

**Comment by a3nima** [comment]
at least I can say in other environments or with other programs, that use cuda I dont have a general problem like this (oobabooga, 2 SD webuis). I hope some time it will be working, but i cannot fix something without knowing what the source of the problem is.

**Comment by a3nima** [comment]
OK I figured out the problem and it's nothing to do with h2gogpt... it's my undervolting of my graphics card. of course I thought it's stable, as it's tested with benchmarks and games don't crush. Also I THOUGHT I had tested it without UV, but probably I did not load the the defaults ,just select them in MSI AB.. soo I'm sorry for all my posts :D I just wanted it to get to work so badly... now it does and I can finally test out models and settings with it. Thanks!
